client/src/components/QueryComponent/QueryComponent.js

```diff
@@ -35,13 +35,38 @@ class QueryComponent extends Component {
       showConfirmDeleteModal: false,
       showConfirmOverwriteModal: false,
       newQueryName: "",
-      isOverlayResultsChecked: false
+      isOverlayResultsChecked: false,
+      multiline: false,
+      multilineHolder: false,
+      shiftPressed: false
     };
   }
 
   componentDidMount() {
     this.setState({ queries: settingsService.queries });
     eventService.subscribeEnvironmentChange(this.clearAfterEnvironmentChange);
+    document.addEventListener("keyup", this.upFunction, false);
+    document.addEventListener("keydown", this.downFunction, false);
+    document.addEventListener("click", this.onFocusLost, false);
```

Comment:
I'd prefer to attach events to the text field control itself rather than globally.

Are there options to access the underlying element on TextField? E.g. https://developer.microsoft.com/en-us/fluentui#/controls/web/textfield --> elementRef? Or is there a way to attach an onKeyUp/onKeyDown to the TextField like we do with controls like Toggle? Plus onBlur to handle focus lost?

Reply:
This Is already been addressed